ibft/storage/store.go

```diff
@@ -203,16 +221,49 @@ func (i *ibftStorage) GetParticipants(identifier convert.MessageID, slot phase0.
 	return operators, nil
 }
 
-func (i *ibftStorage) save(value []byte, id string, pk []byte, keyParams ...[]byte) error {
+func (i *ibftStorage) saveParticipants(txn basedb.ReadWriter, identifier convert.MessageID, slot phase0.Slot, operators []spectypes.OperatorID) error {
+	bytes, err := encodeOperators(operators)
+	if err != nil {
+		return fmt.Errorf("encode operators: %w", err)
+	}
+	if err := i.save(txn, bytes, participantsKey, identifier[:], uInt64ToByteSlice(uint64(slot))); err != nil {
+		return fmt.Errorf("save to DB: %w", err)
+	}
+
+	return nil
+}
+
+func mergeParticipants(existingParticipants, newParticipants []spectypes.OperatorID) []spectypes.OperatorID {
+	seen := make(map[spectypes.OperatorID]struct{})
+
+	for _, operatorID := range existingParticipants {
+		seen[operatorID] = struct{}{}
+	}
+
+	for _, operatorID := range newParticipants {
+		seen[operatorID] = struct{}{}
+	}
+
+	result := make([]spectypes.OperatorID, 0, len(seen))
+	for operatorID := range seen {
+		result = append(result, operatorID)
+	}
+
+	slices.Sort(result)
+
+	return result
+}
```

Comment:
i suggest to use slices.Concat and slices.Compact instead, so this func can be just 2-3 lines

https://pkg.go.dev/slices#Compact